encoding_test.go

```diff
@@ -36,6 +36,31 @@ func TestMakeTypedEncoder(t *testing.T) {
 	}
 }
 
+func TestMakeTypedEncoderByValue(t *testing.T) {
+	expErr := fmt.Errorf("command fooTestObj failed")
+	f := MakeTypedEncoder(func(req *Request, w io.Writer, v fooTestObj) error {
+		if v.Good {
+			return nil
+		}
+		return expErr
+	})
+
+	req := &Request{}
+
+	encoderFunc := f(req)
+
+	buf := new(bytes.Buffer)
+	encoder := encoderFunc(buf)
+
+	if err := encoder.Encode(&fooTestObj{true}); err != nil {
```

Comment:
Shouldn't there be a test that passes fooTestObject by value?

Reply:
So, internally we always pass to Encode by pointer (or should, at least).